extensions/wrapper/clients/typescript-client/src/oauth2/OAuth2ClientCredentials.ts

```diff
@@ -0,0 +1,6 @@
+export interface OAuth2ClientCredentials {
```

Comment:
add some documentation to this as providing this enable the authentication method via the OAuth2 Client Credentials Flow.

This might be obvious, but these classes are really public facing classes, part of a library, part of the entry point of that library, so we should be as clean and helpful as possible